pkg/kubelet/cm/dra/manager.go

```diff
@@ -377,13 +353,17 @@ func (m *ManagerImpl) GetResources(pod *v1.Pod, container *v1.Container) (*Conta
 					return fmt.Errorf("unable to get claim info for claim %s in namespace %s", *claimName, pod.Namespace)
 				}
 
-				claimAnnotations := claimInfo.annotationsAsList()
-				klog.V(3).InfoS("Add resource annotations", "claim", *claimName, "annotations", claimAnnotations)
-				annotations = append(annotations, claimAnnotations...)
-
-				devices := claimInfo.cdiDevicesAsList()
-				klog.V(3).InfoS("Add CDI devices", "claim", *claimName, "CDI devices", devices)
-				cdiDevices = append(cdiDevices, devices...)
+				// As of Kubernetes 1.31, CDI device IDs are not passed via annotations anymore.
+				for _, driverData := range claimInfo.DriverState {
+					for _, device := range driverData.Devices {
+						claimRequestName := claim.Request
+						if claimRequestName == "" || len(device.RequestNames) == 0 || slices.Contains(device.RequestNames, claimRequestName) {
+							for _, cdiDeviceID := range device.CDIDeviceIDs {
+								cdiDevices = append(cdiDevices, kubecontainer.CDIDevice{Name: cdiDeviceID})
+							}
+						}
+					}
+				}
```

Comment:
Can this be added as a function hanging off of `claimInfo` like it used to be?

Reply:
Ack

Reply:
I think the reason why I removed `cdiDevicesAsList` from `claiminfo.go` was because that was the only reason why `k8s.io/kubernetes/pkg/kubelet/container` had to be imported there. This conversion looked to me like something that a consumer of the `claimInfo` should own, not the `claimInfo` implemenation itself. It's only used once.

Anyway, it's back...

Reply:
I'm not sure how complete our testing in kubelet is around this (probably could be better), but a new "must map configs and devices to the right containers" E2E tests which stresses mapping of CDI devices to containers is passing.